Mappers/HttpMapper.php

```diff
@@ -1,165 +1,202 @@
+        return $this->sendHttpRequest(self::PUT, $this->urlPlaceholderRender->updateUrl($objectData), [], $objectData);
     }
 
     /**
-     * @deprecated
-     * @param $action
-     * @param $url
-     * @param $options
-     * @return ArraySerializable|\Modules\LibraryModule\Models\Collection|void
-     * @throws EnergonException
+     * @param string $byPropertyName
+     * @param $propertyValue
+     * @return bool
+     * @throws HttpResourceNotFoundException
+     * @throws InfrastructureException
+     * @throws ValidationException
+     * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    protected function sendHttpRequestForGetStatus($action, $url, $options)
+    public function delete(string $byPropertyName, $propertyValue): bool
     {
-        return $this->getHttpClient()->sendHttpRequestForGetStatus($action, $url, $options);
+        $this->sendHttpRequest(self::DELETE, $this->urlPlaceholderRender->deleteUrl([$byPropertyName => $propertyValue]));
+
+        return true;
     }
 
+
     /**
-     * @param ArraySerializable $object
-     * @param $identifierName
-     * @param $urlName
-     * @return string
+     * @param array $objectData
+     * @return Collection|mixed
+     * @throws HttpResourceNotFoundException
+     * @throws InfrastructureException
+     * @throws ValidationException
+     * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    protected function getUrlWithObjectAndIdentifier(ArraySerializable $object, $identifierName, $urlName)
+    public function updatePatch(array $objectData)
     {
-        $identifier = $object->toArray()[$identifierName];
-        return vsprintf($this->getHttpMapperConfig($urlName), [$identifier]);
+        return $this->sendHttpRequest(self::PATH, $this->urlPlaceholderRender->updateUrl($objectData), [], $objectData);
     }
 
     /**
-     * @param $identifierValue
+     * @param $method
      * @param $url
-     * @return string
+     * @param array $params
+     * @param $body
+     * @return Collection|mixed
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Infrastructure\Models\Http\IllegalHeaderValueException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
      */
-    private function getUrlWithIdentifier($identifierValue, $url)
+    protected function sendHttpRequest($method, $url, $params = [], $body = null)
     {
-        $urlTemplate = $this->getHttpMapperConfig($url);
-        return vsprintf($urlTemplate, [$identifierValue]);
+        if (!$this->headersBuilder->hasHeader('Content-Type')) {
+            $this->headersBuilder->addHeader('Content-Type', self::$defaultContentType);
+        }
+
+        $request = $this->requestFactory->create(
+            $method,
+            $this->buildUrl($url, $params),
+            $this->headersBuilder->build(),
+            $body
+        );
+
+        $response = $this->getHttpClient()->send($request);
+
+        if ($response->getStatusCode() >= 400) {
```

Comment:
PLease check how this in PSR7 because previously if status code was 400 or higher it was excpetion.

Not sure about this strategy, and that you can customize it for different request, just select fast fail or save fail. As I see we need fast fail.

Reply:
selected fast fail